### PR TITLE
feat(rest-api): add support for the theme endpoint to oeq-ts-rest-api

### PR DESCRIPTION
**Pull request by edalex-yinzi**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

Create api with tests for Theme module.

API: 
  - getThemeSettings,
  - updateThemeSettings,
  - getThemeLogo,
  - updateThemeLogo,

Because I am still struggling with tests for theme logo, only contains tests for:
 - getThemeSettings
 - updateThemeSettings

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by edalex-yinzi** [comment]
testAddMyResource failed

File: oeq-ts-rest-api/src/Theme.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * Retrieve theme logo
+ *
+ * @param apiBasePath Base URI to the oEQ institution and API
+ */
+export const getThemeLogo = (apiBasePath: string): Promise<string> =>
+  GET<string>(
+    apiBasePath + THEME_ROOT_PATH + '/newLogo.png',
+    (data): data is any => is<any>(data)
```
Comment (edalex-ian) [inline oeq-ts-rest-api/src/Theme.ts:72]: As discussed, don't use `is` (typescript-is) here, instead you need to manually write a type guard.

File: oeq-ts-rest-api/test/Theme.test.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import { ThemeSettings } from '../src/Theme';
+
+beforeAll(() => OEQ.Auth.login(TC.API_PATH, TC.USERNAME, TC.PASSWORD));
+
+afterAll(() => OEQ.Auth.logout(TC.API_PATH, true));
+
+describe('Theme settings', () => {
+  let settingsAtStart: ThemeSettings;
+
+  beforeAll(async () => {
```
Comment (edalex-ian) [inline oeq-ts-rest-api/test/Theme.test.ts:29]: Be a good idea to add an `afterAll()` to return them to `settingsAtStart`.

File: oeq-ts-rest-api/tsconfig.json [hunk 3]
```diff
@@ -2,7 +2,10 @@
   "include": ["src", "types"],
   "compilerOptions": {
     "module": "ES6",
-    "lib": ["ES6"],
+    "lib": [
+      "ES6",
+      "dom"
```
Comment (edalex-ian) [inline oeq-ts-rest-api/tsconfig.json:7]: Do we really need this?

I gather you've added for `File`, but the oEQ server has no notion of a `dom` `File`. So perhaps we need something else, as that concern would only be in the front-end's Theme Module.

**Review by edalex-ian — APPROVED** [review]
Looks pretty good @edalex-yinzi , just need to:

- Write you own type guard instead of using `is`;
- Tidy up the modification of the settings after test; and
- Revisit the use of `File` and remove `dom` from the libraries.

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Whoops, that was meant to be a 'request changes'. Sorry. :wink:

File: oeq-ts-rest-api/src/Theme.ts (hunk 1 above)
Comment (edalex-yinzi) [reply oeq-ts-rest-api/src/Theme.ts:72]: Yes, just wonder do we still need this API, if we are not going to test the reset of logo API.

File: oeq-ts-rest-api/test/Theme.test.ts (hunk 2 above)
Comment (edalex-yinzi) [reply oeq-ts-rest-api/test/Theme.test.ts:29]: Do you mean reset the value of the setting in afterAll?

Comment (edalex-ian) [reply oeq-ts-rest-api/test/Theme.test.ts:29]: Yup. :+1:

File: oeq-ts-rest-api/src/Theme.ts (hunk 1 above)
Comment (edalex-ian) [reply oeq-ts-rest-api/src/Theme.ts:72]: Could drop it, and just add something like:

```typescript
export const getThemeLogoUrl = (apiBasePath: string): string =>
  apiBasePath + THEME_ROOT_PATH + '/newLogo.png';
```

File: oeq-ts-rest-api/tsconfig.json (hunk 3 above)
Comment (edalex-yinzi) [reply oeq-ts-rest-api/tsconfig.json:7]: Use blob instead?

Comment (edalex-ian) [reply oeq-ts-rest-api/tsconfig.json:7]: Yeah, if that works and matches the server API.

Comment (edalex-yinzi) [reply oeq-ts-rest-api/tsconfig.json:7]: After testing, `Buffer` might be a good type.

Comment (edalex-ian) [reply oeq-ts-rest-api/tsconfig.json:7]: Just make sure whatever you choose works in Browsers.

**Comment by edalex-yinzi** [comment]
I might not implement any test for theme logo in this branch because it needs more consideration. 
Also, I keep the `File` type in theme logo API in order to use it in browsers.

What do you think? @edalex-ian

**Review by edalex-ian — APPROVED** [review]
Sure thing. It's not super critical, and it's not leaving it worse than you found it, so let's proceed.

**Comment by edalex-ian** [comment]
Heh, flaked on `MyResourcesTest`. Good to merge. :wink:

**Pull request merged**